Sources/Filters/General/TubeFilter/index.js

```diff
@@ -692,6 +692,22 @@ function vtkTubeFilter(publicAPI, model) {
       }
     }
 
+    // loop over pointData arrays and resize based on numNewPts
+    const numArrays = output.getPointData().getNumberOfArrays();
+    let oldArray = null;
+    let newArray = null;
+    for (let i = 0; i < numArrays; i++) {
+      oldArray = output.getPointData().getArrayByIndex(0);
```

Comment:
Normally the outPD and outCD should give you some helpers. But those helper are the one who have issue.

For now, you could implement your own array management in the tube filter. But for that you should create the output array based on what you have as input. (as opposed of using the output to find the size that you need to adjust)

Reply:
It needed to be 0 instead of i since it was referencing the output array, which was being modified later in the for loop. I modified it so that the output array is based on the input, as you suggested. As a result, I needed to changed the 0 to i, to accommodate this change. I made a new commit for this change